nibabel/gifti/parse_gifti_fast.py

```diff
@@ -167,14 +167,14 @@ def StartElementHandler(self, name, attrs):
             if "ArrayIndexingOrder" in attrs:
                 self.da.ind_ord = array_index_order_codes.code[
                     attrs["ArrayIndexingOrder"]]
-            if "Dimensionality" in attrs:
-                self.da.num_dim = int(attrs["Dimensionality"])
-            for i in range(self.da.num_dim):
+            self.num_dim = int(attrs.get("Dimensionality", 0))
+            for i in range(self.num_dim):
                 di = "Dim%s" % str(i)
                 if di in attrs:
                     self.da.dims.append(int(attrs[di]))
             # dimensionality has to correspond to the number of DimX given
-            assert len(self.da.dims) == self.da.num_dim
+            # TODO (bcipolli): don't assert; raise parse warning, and recover.
+            assert len(self.da.dims) == self.num_dim
```

Comment:
No problem at all, thanks for your eyes on it! Good point; it was a member variable and I didn't think hard enough to realize it could be made local now. Will take care of that!